modules/tinymce/src/themes/silver/test/ts/headless/components/colorpicker/ColorPickerTest.ts

```diff
@@ -162,4 +164,110 @@ describe('headless.tinymce.themes.silver.components.colorpicker.ColorPickerTest'
     );
 
   });
+
+  context('TINY-9819, force_hex_color option set to default (false)', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+    }, []);
+
+    it('should not affect data-mce-style attribute', () => {
+      const editor = hook.editor();
+      assert.isFalse(editor.options.get('force_hex_color'), 'force_hex_color must be false');
+
+      editor.setContent('<p>colour me</p>');
+      TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 'color me'.length);
+
+      const hexColor = '#E03E2D';
+      editor.execCommand('mceApplyTextcolor', 'forecolor' as any, hexColor);
+
+      const outerHTML = editor.dom.getOuterHTML(editor.getBody());
+      const dataMceStyle = outerHTML.match(/data-mce-style="[^"]+"/)?.[0];
+      assert.isString(dataMceStyle, `Can't find data-mce-style in "${outerHTML}"`);
+      const expectedColor = 'color: rgb(224, 62, 45);';
+      expect(dataMceStyle).to.include(expectedColor, `"${dataMceStyle}" does not include expected color: "${expectedColor}".`);
+    });
```

Comment:
I am wondering if we can do something like this. I am not sure if the selector is exactly right but I believe something like that should work

```suggestion
		TinyAssertions.assertContentPresence(editor, { [`span[data-mce-style="color: ${hexColor}"]`]: 1 });
```

Reply:
Oh that's much nicer, just put that in now